### PR TITLE
Implemented attribute clauses

### PR DESCRIPTION
**Pull request by gareth-robinson**
I've added the ability to include attribute 'where' clauses that are suggested in the README.md but don't appear to be implemented. There are examples in the selection.feature e.g.
"ex:favouriteReads[so:author=Iain M Banks][so:name=Excession] @index"
I ended up reworking most of the internal logic in ld-query.js to do this though.

I've also corrected the 'main' entry in package.json

### CODE REVIEW

File: features/selection.feature [hunk 1]
```diff
@@ -2,60 +2,101 @@ Feature: select values using query syntax
...
-    Scenario: Query for recursive properties
\ No newline at end of file
+    Scenario: Get the first result for a specific author
+      Given I query for "ex:favouriteReads[so:author=Iain M Banks]"
+      When I get the result's json
+      Then the json should match
+        | json                                                                                                                                             |
+        | { "http://schema.org/author": [ { "@value": "Iain M Banks" } ], "http://schema.org/name": [ { "@value": "Excession" } ], "@index": "banks-exc" } |
+
+    Scenario: Query for favourite reads for a specific author
```
Comment (goofballLogic) [inline features/selection.feature:68]: This is an excellent feature, but not one which was originally intended by the concept of attributes. This is something more like an arbitrary query clause.
I suggest that you separate this out into a distinct issue. Attributes like \@index are what was intended for this feature.

Note that all features need to be more-or-less mapped to https://dev.w3.org/2006/webapi/selectors-api2/#interface-definitions

File: features/selection.feature [hunk 2]
```diff
@@ -2,60 +2,101 @@ Feature: select values using query syntax
...
+
+    Scenario: Query for favourite reads for a specific author and title, then get the json
+      Given I query for all "ex:favouriteReads[so:author=Iain M Banks][so:name=Excession]"
+      When I get the json for each result
+      Then the the first json should match
+        | json                                                                                                                                             |
+        | { "http://schema.org/author": [ { "@value": "Iain M Banks" } ], "http://schema.org/name": [ { "@value": "Excession" } ], "@index": "banks-exc" } |
+
+    Scenario: Query for favourite reads for a specific author and title, then get the index
+      Given I query for "ex:favouriteReads[so:author=Iain M Banks][so:name=Excession] @index"
```
Comment (goofballLogic) [inline features/selection.feature:98]: Another new (but welcome) feature. I suggest that you separate this out

**Review by goofballLogic — CHANGES_REQUESTED** [review]
Needs to be more granular. Note also that Travis CI is broke.

**Comment by gareth-robinson** [comment]
Ah, I appear to be using some javascript features that are not available in the release that travis is using

**Comment by gareth-robinson** [comment]
going to close this pull request while I rethink and refactor

**Comment by goofballLogic** [comment]
I'll add a note in the documentation that this library is ES5 compatible without transpiling.

**Pull request closed without merging**